File: core/src/test/java/net/dontdrinkandroot/wicket/css/CompositeCssClassTest.java
package net.dontdrinkandroot.wicket.css;

import org.junit.Assert;
import org.junit.Test;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class CompositeCssClassTest
{
    @Test
    public void testOneElement()
    {
        CssClass cssClass;

        cssClass = new CompositeCssClass(new StringCssClass("asdf"));
        Assert.assertEquals("asdf", cssClass.getClassString());

        cssClass = new CompositeCssClass("asdf");
        Assert.assertEquals("asdf", cssClass.getClassString());
    }

    @Test
    public void testMultipleElements()
    {
        CssClass cssClass;

        cssClass = new CompositeCssClass(new StringCssClass("asdf"), new StringCssClass("qwer"));
        Assert.assertEquals("asdf qwer", cssClass.getClassString());

        cssClass = new CompositeCssClass("asdf", "qwer");
        Assert.assertEquals("asdf qwer", cssClass.getClassString());
    }
}

File: core/src/test/kotlin/net/dontdrinkandroot/wicket/css/CompositeCssClassTest.kt
package net.dontdrinkandroot.wicket.css

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class CompositeCssClassTest {

    @Test
    fun testOneElement() {
        var cssClass: CssClass
        cssClass = CompositeCssClass(StringCssClass("asdf"))
        Assertions.assertEquals("asdf", cssClass.classString)
        cssClass = CompositeCssClass("asdf")
        Assertions.assertEquals("asdf", cssClass.classString)
    }

    @Test
    fun testMultipleElements() {
        var cssClass: CssClass
        cssClass = CompositeCssClass(StringCssClass("asdf"), StringCssClass("qwer"))
        Assertions.assertEquals("asdf qwer", cssClass.classString)
        cssClass = CompositeCssClass("asdf", "qwer")
        Assertions.assertEquals("asdf qwer", cssClass.classString)
    }
}